### PR TITLE
Adding singleton pattern to SplitPDFUtils

### LINKED ISSUE #830: Use `Singleton Design pattern` for Utils classes
Utils classes are present at: (ending with *utils.java)
https://github.com/Swati4star/Images-to-PDF/tree/master/app/src/main/java/swati4star/createpdf/util

For reference:
Use method 3 Lazy initialization from below link for implementation:
https://www.geeksforgeeks.org/java-singleton-design-pattern-practices-examples/

### PR DESCRIPTION
**Pull request by cunhazera**
# Description

Added a singleton pattern to SplitPDFUtils. It's just a try, considering that there many Utils classes.

Fixes #830

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
- [ ] `./gradlew assembleDebug assembleRelease`
- [x] `./gradlew checkstyle`

# Checklist:
- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings

### CODE REVIEW

**Comment by cunhazera** [comment]
@Swati4star oh, ok. Thanks. That helps a lot.

**Comment by Swati4star** [comment]
You can implement the same for other utils classes too

**Comment by cunhazera** [comment]
@Swati4star right. Imma do that ASAP

File: app/src/main/java/swati4star/createpdf/util/SplitPDFUtils.java [hunk 1]
```diff
@@ -30,13 +28,18 @@
     private static final int ERROR_PAGE_RANGE = 2;
     private static final int ERROR_INVALID_INPUT = 3;
 
-    private final Activity mContext;
-    private SharedPreferences mSharedPreferences;
+    private static Activity mContext;
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/util/SplitPDFUtils.java]: Isnt this going to cause memory leaks?

Comment (cunhazera) [reply app/src/main/java/swati4star/createpdf/util/SplitPDFUtils.java]: Why it would? Because the `final` was removed?

Comment (Swati4star) [reply app/src/main/java/swati4star/createpdf/util/SplitPDFUtils.java]: because you made it static

Comment (Swati4star) [reply app/src/main/java/swati4star/createpdf/util/SplitPDFUtils.java]: Remove mContext from constructor & get it as parameter to relevant functions

**Comment by Swati4star** [comment]
Closing due to inactivity. Please feel free to reopen if you continue to work on this.

**Pull request closed without merging**